### PR TITLE
Weighted random webhook choice

### PR DESCRIPTION
**Pull request by anatskiy**
This PR adds the `weight` option, which makes a particular webhook be shown more often than the others.

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/buildapp.py [hunk 1]
```diff
@@ -616,16 +616,16 @@ def populate_api_routes(webapp, app):
...
                           '/api/webhooks/{webhook_type}',
                           controller='webhooks',
-                          action='get_random',
+                          action='get_all_by_type',
                           conditions=dict(method=["GET"]))
 
-    webapp.mapper.connect('get_all_by_type',
-                          '/api/webhooks/{webhook_type}/all',
+    webapp.mapper.connect('get_random',
+                          '/api/webhooks/{webhook_type}/random',
```
Comment (martenson) [inline lib/galaxy/webapps/galaxy/buildapp.py:626]: given its usage probably not that important but this change breaks the existing clients of this API

**Comment by martenson** [comment]
@anatskiy Could you please explain a bit what is the random webhook functionality for? Also what are the changes in `client/yarn.lock`?

**Comment by dannon** [comment]
I wish I had noted this when the feature was first implemented, but choosing a random webhook seems like something that should be implemented on the client side, with the server simply serving the index and specific hooks like you would expect from an API.  Should we step back and do that now?

**Comment by martenson** [comment]
@dannon So what is the random webhook feature for?

**Comment by anatskiy** [comment]
> So what is the random webhook feature for?

@martenson according to the idea, after the tool/workflow execution (if webhooks are enabled), a random webhook is shown - not a specific one.

> Should we step back and do that now?

@dannon honestly, I don't see a problem in getting it from the server. But if you think it should be done on the client side, I'll move it there.

**Comment by bgruening** [comment]
In addition to @anatskiy answer about the random-webhook. You can have multiple webhooks per entry-point. In this case one is picked randomly - and now with a weighted randomness.

**Comment by martenson** [comment]
@anatskiy I understand the functionality, but what is the usecase?

**Comment by hexylena** [comment]
@martenson we want to show some webhooks more than others.

E.g. I have a iframe webhook which shows a random page. It is encountered 1/3rd of the time, and has three possible pages to show. Each page is shown 1/3rd of 1/3rd of the time, when ideally they would be shown equally to the other two webhooks, the three pages + the other two, all shown at 20% of the time.

With this, we can change the percentage of the iframe up to correct for this.

**Comment by martenson** [comment]
@erasche I see, interesting, thanks for the explanation

**Comment by martenson** [comment]
@galaxybot test this

**Comment by jmchilton** [comment]
The failing tool describing tour test cases are probably legitimate problems? Does that plugin need to be updated for these changes?

(I'm with Dannon though - I'd prefer something in the client for choosing a random webhook or with the use case of generating a random page I'd prefer that logic in the webhook itself. But I'm not -1 on this or anything - whatever y'all feel is best - I can't imagine anything but the Galaxy UX consuming this API.)

**Comment by anatskiy** [comment]
@jmchilton sure, I'll think how it can be improved.

**Pull request closed without merging**